src/components/ProfilePage.jsx

```diff
@@ -38,45 +49,32 @@ class UserProfile extends React.Component {
   }
 
   onCancel() {
-    this.props.closeEditableField(this.props.currentlyEditingField);
+    this.props.closeField(this.props.currentlyEditingField);
   }
 
   onEdit(fieldName) {
-    this.props.openEditableField(fieldName);
+    this.props.openField(fieldName);
   }
 
-  onSave(fieldName, value, visibility) {
+  onSave(fieldName) {
     const {
-      value: fieldValue,
-      visibility: fieldVisibility,
+      value,
+      visibility,
     } = this.state[fieldName];
 
-    const valueToSave = value != null ? value : fieldValue;
-    const visibilityToSave = visibility != null ? visibility : fieldVisibility;
-
-    if (valueToSave != null) {
-      this.props.saveProfile(
-        this.props.username,
-        {
-          [fieldName]: valueToSave,
-        },
-        fieldName,
-      );
-    }
+    const data = {};
```

Comment:
This `onSave` code is likely to be further refactored and simplified - again, it currently _works_, but I wouldn't critique it too much in this PR.